orte/mca/schizo/ompi/schizo_ompi.c

```diff
@@ -906,11 +908,34 @@ static int setup_fork(orte_job_t *jdata,
         opal_setenv("OMPI_MCA_orte_hnp_uri", orte_process_info.my_hnp_uri, true, &app->env);
     }
 
-    /* setup yield schedule - do not override any user-supplied directive! */
-    if (oversubscribed) {
-        opal_setenv("OMPI_MCA_mpi_yield_when_idle", "1", false, &app->env);
-    } else {
-        opal_setenv("OMPI_MCA_mpi_yield_when_idle", "0", false, &app->env);
+    ompi_mpi_yield_when_idle = false;
+    (void) mca_base_var_register("ompi", "mpi", NULL, "yield_when_idle",
+                                 "Yield the processor when waiting for MPI communication (for MPI processes, will default to 1 when oversubscribing nodes)",
+                                 MCA_BASE_VAR_TYPE_BOOL, NULL, 0, MCA_BASE_VAR_FLAG_DEF_UNSET,
+                                 OPAL_INFO_LVL_9,
+                                 MCA_BASE_VAR_SCOPE_READONLY,
+                                 &ompi_mpi_yield_when_idle);
+
+    bool set_yield_when_idle = true;
+    int var_index = mca_base_var_find("ompi", "mpi", NULL, "yield_when_idle");
```

Comment:
In addition to what @rhc54 said:

1. My $0.02: make `ompi_mpi_yield_when_idle` level 5 or 6, not 9 (per https://github.com/open-mpi/ompi/wiki/MCAParamLevels).
1. IIRC, `mca_base_var_register()` returns the newly-created index -- I don't think you need to explicitly `(void)`-ignore its return and then `mca_base_var_find()`.